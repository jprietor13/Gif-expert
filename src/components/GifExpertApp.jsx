import React, { useState } from 'react'

const GifExpertApp = () => {
  const categories = ["Bleach", "Dragon Ball Z", "Naruto"];
  const [categorie, setCategorie] = useState(categories);

  const handleAddCategorie = () => {
    const newCategorie = "Samurai X";
    setCategorie([...categories, newCategorie])
  }
  return (
    <div>
      <h2>GifExpertApp</h2>
      <hr />
      <button type="button" onClick={handleAddCategorie}>Agregar</button>
      <ol>
        {categorie.map(category =>
          <li key={category}>{category}</li>
        )}
      </ol>
    </div>
  )
}

export default GifExpertApp;
