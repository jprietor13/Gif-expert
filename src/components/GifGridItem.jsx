import React from 'react';

export const GifGridItem = ({ url, title }) => {
  return (
    <div className="card animate__backInDown">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

