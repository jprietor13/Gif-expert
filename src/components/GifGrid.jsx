import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifData } from '../helpers/getGifData';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifData(category)
      .then(images => setImages(images))
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="grid-card">
        {images.map(item =>
          <GifGridItem key={item.id} images={item} />
        )}
      </div>
    </>
  )
}

