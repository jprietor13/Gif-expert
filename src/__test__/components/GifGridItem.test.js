import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe("Pruebas en GifGridItem", () => {
  const title = "title categorie";
  const url = "https://localhost:3000/title_categorie.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe mostrar el componente", () => {
    expect(wrapper).toMatchSnapshot();
  })

  test("Debe tener un parrafo con el titulo", () => {
    const parrafo = wrapper.find('p');
    expect(parrafo.text().trim()).toBe(title);
  });

  test("Debe tener el url y el titulo de la imagen igual al de los props", () => {
    const img = wrapper.find('img');
    //console.log(img.prop('src'))
    expect(img.prop('src')).toBe(url); //props sirve para ver los atributos del elemento html
    expect(img.prop('alt')).toBe(title);
  });

  test("Debe tener el div padre la clase animate__backInDown", () => {
    const div = wrapper.find('div');
    expect(div.prop('className')).toContain('animate__backInDown')
  });
})