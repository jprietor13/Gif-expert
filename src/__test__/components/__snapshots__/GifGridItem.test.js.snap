// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el componente 1`] = `
<div
  className="card animate__backInDown"
>
  <img />
  <p />
</div>
`;
